pkg/client/redhatsso/client_test.go

```diff
@@ -0,0 +1,938 @@
+package redhatsso
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/test/mocks"
+	. "github.com/onsi/gomega"
+	"github.com/patrickmn/go-cache"
+	serviceaccountsclient "github.com/redhat-developer/app-services-sdk-go/serviceaccounts/apiv1internal/client"
+)
+
+const (
+	accountName        = "serviceAccount"
+	accountDescription = "fake service account"
+)
+
+func CreateServiceAccountForTests(accessToken string, server mocks.RedhatSSOMock, accountName string, accountDescription string) serviceaccountsclient.ServiceAccountData {
+	c := &rhSSOClient{
+		configuration: &serviceaccountsclient.Configuration{
+			DefaultHeader: map[string]string{
+				"Authorization": fmt.Sprintf("Bearer %s", accessToken),
+				"Content-Type":  "application/json",
+			},
+			UserAgent: "OpenAPI-Generator/1.0.0/go",
+			Debug:     false,
+			Servers: serviceaccountsclient.ServerConfigurations{
+				{
+					URL: fmt.Sprintf("%s/auth/realms/redhat-external", server.BaseURL()),
+				},
+			},
+		},
+	}
+	serviceAccount, _ := c.CreateServiceAccount(accessToken, accountName, accountDescription)
+	return serviceAccount
+}
+
+func TestNewSSOClient(t *testing.T) {
+	type args struct {
+		config      *keycloak.KeycloakConfig
+		realmConfig *keycloak.KeycloakRealmConfig
+	}
+	tests := []struct {
+		name string
+		args args
+		want SSOClient
+	}{
+		{
+			name: "should pass if client is not nil",
+			args: args{
+				config:      &keycloak.KeycloakConfig{},
+				realmConfig: &keycloak.KeycloakRealmConfig{},
+			},
+			want: &rhSSOClient{
+				config:      &keycloak.KeycloakConfig{},
+				realmConfig: &keycloak.KeycloakRealmConfig{},
+			},
+		},
+	}
+	g := NewWithT(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g.Expect(NewSSOClient(tt.args.config, tt.args.realmConfig)).To(Not(BeNil()))
```

Comment:
`want` is not used. I believe it could be useful to add some check that the created object actually contanis the config you passed in to the NewSSOClient method,